### PR TITLE
* #110: Added support for scala projects release on GitHub.

### LINKED ISSUE #110: Add support for Scala releases on GitHub
## Problem

We currently only support Java projects.

We want to add support for releasing Scala projects on GitHub

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #110

### CODE REVIEW

File: src/main/java/com/exasol/releasedroid/github/GitHubPlatformValidator.java [hunk 1]
```diff
@@ -17,33 +17,36 @@
...
         LOGGER.fine("Validating GitHub-specific requirements.");
         final Report report = Report.validationReport();
-        final String version = repository.getVersion();
-        final ReleaseLetter releaseLetter = repository.getReleaseLetter(version);
-        report.merge(validateChangesFile(repository, releaseLetter));
-        report.merge(validateFileExists(repository, GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
+        final String version = this.repository.getVersion();
+        final ReleaseLetter releaseLetter = this.repository.getReleaseLetter(version);
+        report.merge(validateChangesFile(releaseLetter));
+        report.merge(validateFileExists(this.repository, GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/github/GitHubPlatformValidator.java:41]: I would suggest to remove this parameter, as we now have access to the repo.
```suggestion
        report.merge(validateFileExists(GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
```

File: src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java [hunk 2]
```diff
@@ -18,13 +16,18 @@
...
+    }
 
     @Override
     // [impl->dsn~validate-maven-release-workflow-exists~1]
-    public Report validate(final Repository repository) {
+    public Report validate() {
         final Report report = Report.validationReport();
-        report.merge(validateFileExists(repository, MAVEN_WORKFLOW_PATH, "Workflow for a Maven release."));
-        report.merge(validateMavenPom(repository.getMavenPom()));
+        report.merge(validateFileExists(this.repository, MAVEN_WORKFLOW_PATH, "Workflow for a Maven release."));
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java:29]: remove this parameter as we have it already as instance variable
```suggestion
        report.merge(validateFileExists(MAVEN_WORKFLOW_PATH, "Workflow for a Maven release."));
```

File: src/main/java/com/exasol/releasedroid/repository/BaseRepository.java [hunk 3]
```diff
@@ -156,16 +39,31 @@ public ReleaseLetter getReleaseLetter(final String version) {
...
+    }
+
+    @Override
+    public boolean isOnDefaultBranch() {
+        return this.repositoryGate.isOnDefaultBranch();
+    }
+
+    @Override
+    public String getBranchName() {
+        return this.repositoryGate.getBranchName();
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/repository/BaseRepository.java:67]: We need to think about all this delegation. Please create a ticket (unless there's one already) to address this situation

File: doc/changes/changes_0.4.0.md [hunk 4]
```diff
@@ -10,6 +10,8 @@ Code name:
 
 ## Features
 
+* #110: Added support for scala projects release on GitHub.
```
Comment (chiaradiamarcelo) [inline doc/changes/changes_0.4.0.md:13]:
```suggestion
* #110: Added support for Scala projects release on GitHub.
```

File: src/main/java/com/exasol/releasedroid/github/GitHubPlatformValidator.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/github/GitHubPlatformValidator.java:41]: validateFileExists() method is in the abstract class. In the implementation, we have different types of repositories: Java, Scala. So I also can't move the Repository field to the abstract class.

File: src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/maven/MavenPlatformValidator.java:29]: The same here

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**